src/internet_identity/src/main.rs

```diff
@@ -459,13 +468,22 @@ fn authenticated_anchor_operation<R>(
 
 /// Checks if the caller is authenticated against the anchor provided and returns a reference to the device used.
 /// Traps if the caller is not authenticated.
-fn trap_if_not_authenticated(anchor: &Anchor) -> &Device {
+fn trap_if_not_authenticated(anchor_number: AnchorNumber, anchor: &Anchor) -> &Device {
```

Comment:
this opens the door to errors (i.e. mismatched `anchor_number` and `anchor`), how about fetching the `Anchor` in here instead of passing it as an argument?

Reply:
Makes, sense -> refactored.